UnityProject/Assets/Scripts/Input System/MouseInputController.cs

```diff
@@ -44,26 +45,35 @@ public class MouseInputController : MonoBehaviour
 
 	private Vector3 MouseWorldPosition => Camera.main.ScreenToWorldPoint(CommonInput.mousePosition);
 
+	// last mouse position used in CheckHover() to determine if player hovers other tile
+	private Vector3Int lastMouseWordlPositionInt;
+
 	/// <summary>
 	/// currently triggering aimapply interactable - when mouse is clicked down this is set to the
 	/// interactable that was triggered, then it is re-triggered continuously while the button is held,
 	/// then set back to null when the button is released.
 	/// </summary>
 	private IBaseInteractable<AimApply> triggeredAimApply;
 
+	/// <summary>
+	/// Prefab used to visualise cable placement
+	/// </summary>
+	[SerializeField] private GameObject cableVisualisation;
```

Comment:
I think this is intended to be a generic class, which shouldn't have item/object specific logic in it looking at how it's structured. Placing this in here now permanently couples it to cables and in the future it'll become hard to maintain as more and more pieces are put in, especially as it now has very specific variables, classes, and logic in it.

I think it'd be good to try and move all the specific cable references elsewhere - but would be good to get someone who knows the code base better than me to weight in.

Reply:
I have transferred this functionality to a new script (CablePlacementVisualisation)